src/components/TextInput.jsx

```diff
@@ -114,32 +114,21 @@ export default class TextInput extends Component {
 
     if (!this.props.allowNewLine && value.match(newLineRegex) !== null) {
       value = event.target.value.replace(newLineRegex, ' ');
-
-      // controlled textarea must have value
```

Comment:
@jpuri Did you got my summary of the feedback from Sebastian Markbåge

I feel like we should get rid of 

```
const domNode = React.findDOMNode(this);
domNode.value = value;
```

and instead just do this.props.onChange({value: value});